lib/fog/sakuracloud/models/volume/disk.rb

```diff
@@ -20,8 +20,12 @@ def delete
         alias_method :destroy, :delete
 
         def save
-          requires :name, :plan, :source_archive
-          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
+          requires :name, :plan
+          options = {
+            :source_archive=>@attributes[:source_archive],
+            :size_mb=>@attributes[:size_mb]
+          } 
+          data = service.create_disk(@attributes[:name], @attributes[:plan], options).body["Disk"]
```

Comment:
You don´t need to use the instance variable `@attributes`. Each attribute has its own getter with the same signature of the attribute so instead of `@attributes[:name]` you can use just `name`.